src/data/fintype.lean

```diff
@@ -428,6 +428,33 @@ finset.subtype.fintype s
 @[simp] lemma fintype.card_coe (s : finset α) :
   fintype.card (↑s : set α) = s.card := card_attach
 
+lemma finset.card_le_one_iff {s : finset α} :
+  s.card ≤ 1 ↔ ∀ {x y}, x ∈ s → y ∈ s → x = y :=
+begin
+  let t : set α := ↑s,
+  letI : fintype t := finset_coe.fintype s,
+  have : fintype.card t = s.card := fintype.card_coe s,
+  rw [← this, fintype.card_le_one_iff],
+  split,
+  { assume H x y hx hy,
+    exact subtype.mk.inj (H ⟨x, hx⟩ ⟨y, hy⟩) },
+  { assume H x y,
+    exact subtype.eq (H x.2 y.2) }
+end
+
+lemma finset.one_lt_card_iff {s : finset α} :
+  1 < s.card ↔ ∃ x y, (x ∈ s) ∧ (y ∈ s) ∧ x ≠ y :=
+begin
+  classical,
+  rw ← not_iff_not,
+  push_neg,
+  rw finset.card_le_one_iff,
```

Comment:
Shouldn't `simpa` be able to close the goal instead of this `rw`? You might need to add some lemmas to the `simp`-set that `push_neg` also used.
If I understand correctly, the only reason you can't say `exact finset.card_le_one_iff` is because the `simp`-step in `push_neg` did "too much". But a `simpa` should be able to apply the same level of "too much" to `finset.card_le_one_iff` and then `exact` it.

Reply:
Indeed. Done. Thanks!